engine/common/insight_log/resource_insight_logs.py

```diff
@@ -0,0 +1,132 @@
+        """
+        insight_logs_verbal returns resource name for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_type_verbal(self) -> str:
+        """
+        insight_logs_type_verbal resource type for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_serialized(self) -> dict:
+        """
+        insight_logs_serialized returns resource, serialized for insight_log
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def insight_logs_metadata(self) -> dict:
+        """
+        insight_logs_metadata returns resource metadata which should be present in the insight_log line
+        """
+        pass
+
+
+def resource_insight_log(instance: InsightLoggable, author, event: EntityEvent, prev_state=None, new_state=None):
+    try:
+        organization = author.organization
+        DynamicSetting = apps.get_model("base", "DynamicSetting")
+        org_id_to_enable_insight_logs, _ = DynamicSetting.objects.get_or_create(
+            name="org_id_to_enable_insight_logs",
+            defaults={"json_value": []},
+        )
+        log_all = "all" in org_id_to_enable_insight_logs.json_value
+        insight_logs_enabled = organization.id in org_id_to_enable_insight_logs.json_value
+        if insight_logs_enabled or log_all:
+            tenant_id = organization.stack_id
+            author_id = author.public_primary_key
+            author = json.dumps(author.username)
+            entity_type = instance.insight_logs_type_verbal
+            try:
+                entity_id = instance.public_primary_key
+            except AttributeError:
+                entity_id = instance.id  # Fallback for entities which have no public_primary_key
+            entity_name = json.dumps(instance.insight_logs_verbal)
+            metadata = instance.insight_logs_metadata
+            log_line = f"tenant_id={tenant_id} author_id={author_id} author={author} action_type=resource action={event.value} resource_type={entity_type} resource_id={entity_id} resource_name={entity_name}"  # noqa
+            for k, v in metadata.items():
+                log_line += f" {k}={json.dumps(v)}"
+            if prev_state and new_state:
+                prev_state, new_state = state_diff_finder(prev_state, new_state)
+                prev_state = escape_json_str_for_insight_log(json.dumps(format_state_for_insight_log(prev_state)))
+                new_state = escape_json_str_for_insight_log(json.dumps(format_state_for_insight_log(new_state)))
+                log_line += f' prev_state="{prev_state}"'
+                log_line += f' new_state="{new_state}"'
+            insight_logger.info(log_line)
+    except Exception as e:
+        logger.warning(f"insight_log.failed_to_write_entity_insight_log exception={e}")
+
+
+def state_diff_finder(before: dict, after: dict):
+    """
+    state_diff_finder finds diff between two serialized representations of the resource
```

Comment:
can you please clarify what is before_diff and after_diff?
what was removed and what was added?